src/gt4py/next/otf/recipes.py

```diff
@@ -21,3 +22,17 @@ class OTFCompileWorkflow(workflow.NamedStepSequence):
     bindings: workflow.Workflow[stages.ProgramSource, stages.CompilableSource]
     compilation: workflow.Workflow[stages.CompilableSource, stages.CompiledProgram]
     decoration: workflow.Workflow[stages.CompiledProgram, stages.CompiledProgram]
+
+    def __post_init__(self) -> None:
+        from gt4py.next import metrics
+
+        if config.COLLECT_METRICS:
+            for key in dataclasses.fields(self):
+                # TODO we don't know the program name...
+                object.__setattr__(
+                    self,
+                    key.name,
+                    workflow.TimedStep(
+                        metrics.global_metric_container["foo"][key.name], getattr(self, key.name)
```

Comment:
also this should probably go to a different container and doesn't need more than one entry...